src/TTextEdit.cpp

```diff
@@ -1425,7 +1425,11 @@ QString TTextEdit::getSelectedText(const QChar& newlineChar)
 {
     // mPA QPoint where selection started
     // mPB QPoint where selection ended
-
+    // try to prevent crash if buffer is batch deleted
+    if (mPA.y() > mpBuffer->lineBuffer.size() - 1 || mPB.y() > mpBuffer->lineBuffer.size() - 1){
```

Comment:
Might I suggest replacing the `if (` with a `while (` if the buffer size is really small and there is a lot of long lines coming in, doing this once *might* not be enough...

Reply:
sure :+1: 
I think you are more experienced with this part of code and know what is going on here. 
For me it was more shooting in the dark and it seemed to work but we need surely look out for other issues this could create.

Reply:
I did, in the past try to insert an event generator that would fire every time the `TBuffer::shrinkbuffer(...)` method was called for any `TConsole` so that user scripts/packages that kept references (indexes) to line numbers could have a chance to know that they needed to remove a corresponding number of lines from those indexes into a particular console if it did get shrunk. However, IIRC it didn't work as expected and got dropped - I think it might be useful to resurrect it if possible. Does anyone else have any thoughts about that?